nle/nethack/nethack.py

```diff
@@ -154,7 +154,7 @@ def _write_wizkit_file(self, wizkit_items):
         # TODO ideally we need to check the validity of the requested items
         with open(os.path.join(self._vardir, WIZKIT_FNAME), "w") as f:
             for item in wizkit_items:
-                f.write(f"{item}\n")
+                f.write("{}\n".format(item))
```

Comment:
Idiosyncratic weird take  from @heiner: Please don't ever use `str.format`. It was a mistake in every sense. (Obviously the f string here was unnecessary too, that slipped through my review orf @yobibyte's code.)

In this case here, let's use either `f.write("%s\n" % item)` or indeed `f.write(item)` followed by `f.write("\n")`.

Reply:
What's bad about f strings? Slow?

Reply:
isn't string.format considered the more pythonic, and the C-style printf considered old hat?

Reply:
I remember moving across from some reason about the time I moved from py2.7 to 3. I think its to do with the fact that [its considered quirkier](https://docs.python.org/3/library/stdtypes.html#printf-style-string-formatting), and str.format was supposed to be formatting "done right".  

TBH i have no strong preferences either way because im not really aware of the great advantages either way, but I liked `str.format` for the named arguments that you can do. Very keen to hear whats behind the ideosyncratic take and am ultimately flexible.

Reply:
@rockt It's because f strings are py3.6+ and I think the only thing stopping us supporting 3.5, which is default on ubuntu 16.04

Reply:
@rockt f-strings actually are great (and generally faster than other formatting systems in python), but they require python 3.6+.

My only issue with str.format is that they are fairly slow, and C-style formatting is slightly faster on average, so ¯\_(ツ)_/¯ (though f-strings a couple of times faster IIRC still).

Reply:
f-strings are fine but not supported on older versions of Python 3.

`str.format` was just a mistake, made worse by a superstitious belief that it's the true / only way of doing this in Python3. `%` is better in almost every way. It can do named arguments as well if you want it to: `%(ham)s, %(spam)i" % dict(spam=42, ham="eggs")`.

Reply:
```suggestion
                f.write("%s\n" % item)
```

Reply:
Fine by me :)

(Although for files it's perhaps ever so slightly more efficient to just write one string after another.)